pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go

```diff
@@ -457,19 +487,48 @@ func (r *ReconcileLinstorSatelliteSet) reconcileAutomaticDeviceSetup(ctx context
 		}
 
 		for _, dev := range ourNode {
-			paths = append(paths, dev.Device)
+			emptyDevices.Insert(dev.Device)
+		}
+	}
+
+	logger.WithField("emptyDevices", emptyDevices).Debug("got available devices")
+
+	for _, pool := range nodeSet.Spec.StoragePools.All() {
+		logger := logger.WithField("pool", pool)
+
+		logger.Debug("checking configuration for storage pool")
+
+		if !emptyDevices.HasAny(pool.GetDevicePaths()...) {
+			logger.Debug("no device to configure, skipping")
+			continue
 		}
+
+		if !emptyDevices.HasAll(pool.GetDevicePaths()...) {
+			return fmt.Errorf("failed to prepare storage devices for pool '%s' on node '%s': not all devices present and empty", pool.GetName(), pod.Spec.NodeName)
+		}
+
+		err := r.linstorClient.Nodes.CreateDevicePool(ctx, pod.Spec.NodeName, pool.ToPhysicalStorageCreate())
+		if err != nil {
+			return err
+		}
+
+		emptyDevices.Delete(pool.GetDevicePaths()...)
 	}
 
-	for _, path := range paths {
-		logger.WithField("path", path).Debug("prepare device")
+	logger.Debug("finished setting up devices for storage pool")
+
+	// Skip setting up remaining devices
+	if nodeSet.Spec.AutomaticStorageType == automaticStorageTypeNone {
+		return nil
+	}
 
+	for emptyDevice := range emptyDevices {
 		// Note: not found returns -1, so in this case name == path, which is exactly what we want
-		idx := strings.LastIndex(path, "/")
-		name := path[idx+1:]
+		idx := strings.LastIndex(emptyDevice, "/")
+		name := "dev-" + emptyDevice[idx+1:]
```

Comment:
I would find it strange to have a VG named `dev-sda`, but I'm not sure of a better alternative :upside_down_face: 

`pool-sda`? `vg-sda` (doesn't work for ZFS)?

Reply:
`autopool-sda`?

Reply:
> `autopool-sda`?

Fine by me.